nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-11-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/TestPublisherPool.java

```diff
@@ -65,5 +70,40 @@ public void testPoisonedLeaseNotReturnedToPool() {
         lease.close();
         assertEquals(0, pool.available());
     }
+    @Test
+    public void testUseTransactionWithUUID() {
+        final Map<String, Object> kafkaProperties = new HashMap<>();
+        kafkaProperties.put("bootstrap.servers", "localhost:1111");
+        kafkaProperties.put("key.serializer", ByteArraySerializer.class.getName());
+        kafkaProperties.put("value.serializer", ByteArraySerializer.class.getName());
 
+        final PublisherPool pool = new PublisherPool(kafkaProperties, Mockito.mock(ComponentLog.class), 1024 * 1024, 1000L, true, null, null, StandardCharsets.UTF_8);
+        PublisherLease publisherLease = pool.obtainPublisher();
+
+        Producer producer = (Producer) Whitebox.getInternalState(publisherLease, "producer");
+        TransactionManager transactionManager = (TransactionManager) Whitebox.getInternalState(producer, "transactionManager");
+        String transactionalId = (String) Whitebox.getInternalState(transactionManager, "transactionalId");
+        try {
+            UUID uuid = UUID.fromString(transactionalId);
```

Comment:
As there's no real magic to a UUID here, just important that a value be set, would recommend we just assert that the value is non-null, instead of asserting that it is a valid UUID.